usr/share/rear/conf/default.conf

```diff
@@ -803,14 +803,24 @@ CLONE_ALL_USERS_GROUPS="no"
 # users and groups there and not with the users and groups of the original system
 # except those users and groups that get copied into the ReaR recovery system.
 
-# SSH_ROOT_PASSWORD defines a root password to allow SSH connection whithout a public/private key pair
+# SSH_ROOT_PASSWORD defines a password for remote access to the recovery system as 'root' via SSH whithout requiring a
+# public/private key pair. This password is valid only while the recovery system is running and will not allow access
+# to a restored target system.
 # Be aware, the password is saved in hashed MD5 format (do not forget the password after months:)
 # Generate a hashed password with the following command:
 #   echo "my_rescue_root_password" | openssl passwd -1 -stdin
 # and copy paste the output of openssl to variable SSH_ROOT_PASSWORD='...' (mind the single quotes!)
 # into config file /etc/rear/local.conf
 SSH_ROOT_PASSWORD=
 
+# SSH_UNPROTECTED_PRIVATE_KEYS=yes makes ReaR also include SSH keys without a passphrase in the rescue
+# media. This is not recommended because ReaR - by default - builds a rescue media that does not contains
+# any secrets from the source system.
+# Instead of enabling this consider creating additional SSH keys with passphrases to be used only during
+# recovery. You can even have the same key, once with and once without passphrase. The one without will be
+# used during regular operations and the one with passphrase on the rescue system.
+SSH_UNPROTECTED_PRIVATE_KEYS=no
```

Comment:
I'd set default value on YES, otherwise we can introduce some nasty surprise for admins, which will not be widely appreciated. ...

Reply:
In general yes, specifically here we should aim for a breaking change in order to increase the default security. Admins who need this behavior will have to update their configuration.

Reply:
IMHO this is not "setup-and-forget" approach...

Reply:
If the recovery relies on SSH keys then in any case the admin should test the setup.

I think that this change is similar to the introduction of a root password for the rescue system (misnamed `SSH_ROOT_PASSWORD`) which has to be explicitly set. FWIW, I also was against that feature but let myself be convinced by somebody having a legitimate use case. At least we also support encrypted passwords.

For me there are now two alternatives and we have to choose one:

1. keep up the definition of the rescue image as being **secure by default**, which mostly means that it doesn't contain secrets.
1. change our definition of the rescue image and declare it as **secret** that has to be kept locked away like the backup.

Which one do you prefer?

Reply:
>I think that this change is similar to the introduction of a root password for the rescue system (misnamed SSH_ROOT_PASSWORD) which has to be explicitly set

If you are talking about 6156f2fcbfaeccc239d930fe60d93fc7f4f4b522, it is not similar. That commit introduced new functionality, but here we are changing existing functionality on which people relay ... 

> For me there are now two alternatives and we have to choose one:
> 
> 1. keep up the definition of the rescue image as being secure by default, which mostly means that it doesn't contain secrets.
> 2. change our definition of the rescue image and declare it as secret that has to be kept locked away like the backup.

I'd say option 2. is the right one. ISO must be kept safe any time. If you have it compromised, one can't guarantee integrity of restored system any more.

Reply:
Let's take this discussion to #1511 please

Reply:
@schlomo Would you mind including an additional option for not copying private keys at all? Private key protection may not be strong enough to withstand an offline attack.

Reply:
Interesting thought. How should users restore their backup if the keys are required for that? Or do you think about use cases where the backup retrieval is not based on SSH and you want to avoid exposing SSH keys via the rescue media?

If that is the case then I would introduce a new variable `SSH_NO_KEYS_IN_RESCUE_MEDIA` that would drop both the host and client keys. Challenge here is to create new SSH host keys as the SSH server doesn't start without.

Reply:
My primary use case here is less-guarded environments like home offices or branch offices. I usually cannot really protect rescue media there, although I can protect systems via disk encryption (including boot disk encryption, e.g. via [TCG's OPAL standard](https://github.com/sedutil/sedutil)).

And I actually use SSH to connect to the backup server and initially wanted to avoid entering passwords during restore, too. But given the implications, I'd prefer the inconvenience of entering a password once or twice during restore.

Cf.  https://github.com/rear/rear/issues/1512#issuecomment-331638066:
> Connecting via SSH from the rescue system to other hosts (e.g. for restoring a backup) will generally require the user to enter a password, as SSH public/private key authentication will not be established.

And:
> (Public/private key authentication is more suited for long-term trust relations between secure systems. It is ill-suited for temporary trust relations like the one involving a rescue system as public/private key authentication has to be explicitly revoked, which can easily fail or be forgotten.)

Introducing `SSH_NO_KEYS_IN_RESCUE_MEDIA` would seem perfect for me. And yes, re-generating SSH host keys would be required. I've used the following code over here to do that in the build stage:
```
# Generate host key for incoming SSH which was excluded from backup by prep/INTERNAL/default/205_secure_ssh_private_keys.sh
Log "Generating SSH host key"
ssh-keygen -t ed25519 -N '' -f "$ROOTFS_DIR/etc/ssh/ssh_host_ed25519_key"
```

Even better: Re-generate SSH host keys on each rescue system boot, then display their fingerprints on the console to be verified by the admin when logging into the rescue system via SSH.

With SSH host keys re-generated for the individual rescue medium, attacks would be limited to the time where such rescue system is actually running. There would be no long-term threat to the original system or the restored system.

With SSH host keys re-generated on each boot, an attacker would have to be present during such boot and then have an administrator log into the rescue system.